libcloudforensics/providers/gcp/internal/compute.py

```diff
@@ -697,6 +697,97 @@ def ImportImageFromStorage(self,
             storage_image_path, image_name))
     return GoogleComputeImage(self.project_id, '', image_name)
 
+  def InsertFirewallRule(self, body: Dict[str, Any]) -> None:
+    """Insert a firewall rule to the project.
+
+    Args:
+      body (Dict): The request body.
+          https://googleapis.github.io/google-api-python-client/docs/dyn/compute_v1.firewalls.html#insert  # pylint: disable=line-too-long
+    """
+
+    logger.info( 'Inserting firewall rule {0:s}, '
+            'targeting tags: {1!s}.'.format(body['name'], body['targetTags'] ))
+    firewall_client = self.GceApi().firewalls()
+    request = firewall_client.insert(project=self.project_id, body=body)
+    response = request.execute()
+    self.BlockOperation(response)
+
+  def AddDenyAllFirewallRules(self,
```

Comment:
I think this function would be better placed in `forensics.py`, as it contains mostly "DFIR logic"

Reply:
Done